File: app/src/main/java/com/cottondroid/todoachieved/task/list/TaskListPresenter.java
package com.cottondroid.todoachieved.task.list;

import com.cottondroid.todoachieved.task.TaskRepository;
import com.cottondroid.todoachieved.task.model.TodoTask;
import com.cottondroid.todoachieved.task.network.TaskUpdateListener;

import javax.inject.Inject;
import java.util.List;

import io.reactivex.Flowable;

public class TaskListPresenter {
    private final TaskRepository taskRepository;
    private final TaskUpdateListener taskListener;

    @Inject
    public TaskListPresenter(TaskRepository taskRepository,
                             TaskUpdateListener taskListener) {
        this.taskRepository = taskRepository;
        this.taskListener = taskListener;
    }

    public Flowable<List<TodoTask>> getItems() {
        return taskRepository.getTasks();
    }

    public void onResume() {
        taskRepository.registerToTaskUpdates(taskListener);
    }

    public void onLogin() {
        taskRepository.registerToTaskUpdates(taskListener);
    }

    public void onPause() {
        taskRepository.unregisterFromTaskUpdates(taskListener);
    }
}

File: app/src/main/java/com/cottondroid/todoachieved/task/list/TaskListPresenter.kt
package com.cottondroid.todoachieved.task.list

import com.cottondroid.todoachieved.task.TaskRepository
import com.cottondroid.todoachieved.task.model.TodoTask
import com.cottondroid.todoachieved.task.network.TaskUpdateListener
import io.reactivex.Flowable
import javax.inject.Inject

class TaskListPresenter @Inject constructor(
        private val taskRepository: TaskRepository,
        private val taskListener: TaskUpdateListener) {
    val items: Flowable<List<TodoTask>>
        get() = taskRepository.tasks

    fun onResume() {
        taskRepository.registerToTaskUpdates(taskListener)
    }

    fun onLogin() {
        taskRepository.registerToTaskUpdates(taskListener)
    }

    fun onPause() {
        taskRepository.unregisterFromTaskUpdates(taskListener)
    }

}